extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/predictor/FilterService.java

```diff
@@ -0,0 +1,124 @@
+            Double threshold) {
+
+        final OrtSession.Result results;
+        try {
+            final String[][] throttlingInferenceRows = convertToArray(throttlingMessages);
+            results = onnxModelRunner.runModel(throttlingInferenceRows);
+            return processModelResults(results, throttlingMessages, threshold);
+        } catch (OrtException e) {
+            throw new PreBidException("Exception during model inference: ", e);
+        }
+    }
+
+    private static String[][] convertToArray(List<ThrottlingMessage> messages) {
+        return messages.stream()
+                .map(message -> new String[]{
+                        message.getBrowser(),
+                        message.getBidder(),
+                        message.getAdUnitCode(),
+                        message.getCountry(),
+                        message.getHostname(),
+                        message.getDevice(),
+                        message.getHourBucket(),
+                        message.getMinuteQuadrant()})
+                .toArray(String[][]::new);
+    }
+
+    private Map<String, Map<String, Boolean>> processModelResults(
+            OrtSession.Result results,
+            List<ThrottlingMessage> throttlingMessages,
+            Double threshold) {
+
+        validateThrottlingMessages(throttlingMessages);
+
+        return StreamSupport.stream(results.spliterator(), false)
+                .filter(onnxItem -> {
+                    validateOnnxTensor(onnxItem);
+                    return Objects.equals(onnxItem.getKey(), "probabilities");
+                })
+                .map(onnxItem -> (OnnxTensor) onnxItem.getValue())
+                .map(tensor -> {
+                    validateTensorSize(tensor, throttlingMessages.size());
+                    return extractAndProcessProbabilities(tensor, throttlingMessages, threshold);
+                })
+                .flatMap(map -> map.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private void validateThrottlingMessages(List<ThrottlingMessage> throttlingMessages) {
+        if (throttlingMessages == null || CollectionUtils.isEmpty(throttlingMessages)) {
+            throw new PreBidException("throttlingMessages cannot be null or empty");
+        }
+    }
+
+    private void validateOnnxTensor(Map.Entry<String, OnnxValue> onnxItem) {
+        if (!(onnxItem.getValue() instanceof OnnxTensor)) {
+            throw new PreBidException("Expected OnnxTensor for 'probabilities', but found: "
+                    + onnxItem.getValue().getClass().getName());
+        }
+    }
+
+    private void validateTensorSize(OnnxTensor tensor, int expectedSize) {
+        final long[] tensorShape = tensor.getInfo().getShape();
+        if (tensorShape.length == 0 || tensorShape[0] != expectedSize) {
+            throw new PreBidException("Mismatch between tensor size and throttlingMessages size");
+        }
+    }
```

Comment:
Please, add `static` modifier for methods declaration.

Reply:
added static modifier for `validateThrottlingMessages`, `validateOnnxTensor`, `validateTensorSize`